src/stirling/source_connectors/socket_tracer/socket_trace_connector.cc

```diff
@@ -617,6 +627,30 @@ void SocketTraceConnector::UpdateTrackerTraceLevel(ConnTracker* tracker) {
   }
 }
 
+// Verifies that our openssl tracing does not encounter conditions that invalidate our
+// assumptions and records error conditions in prometheus metrics.
+void SocketTraceConnector::CheckTracerState() {
+  // The check that state is not uninitialized is required for socket_trace_connector_test.
+  // Since it doesn't initialize the BPF program, accessing the map will fail.
+  if (state() == State::kUninitialized) {
+    return;
+  }
+
+  int error_code;
+  openssl_trace_state_->get_value(kOpenSSLTraceStatusIdx, error_code);
+
+  if (error_code == kOpenSSLMismatchedFDsDetected) {
+    openssl_trace_mismatched_fds_counter_.Increment();
```

Comment:
This seems like it will undercount mismatched fds, if they occur more than once in between calls to this func. I think you said we don't expect this to ever happen so maybe its fine if we undercount, but just want to double check.

Reply:
Undercounting is fine. We just need to be able to detect if it happens despite our expectation that it should never occur (assuming our understanding of the approach taken in #1123 is correct). Appreciate the thoughtful review 👍